media/open_zim.ksy

```diff
@@ -0,0 +1,380 @@
+            type: u4
+            doc: (optional) identifies a revision of the contents of this directory entry, needed to identify updates or revisions in the original history
+          - id: body
+            type:
+              switch-on: mime_idx == 0xffff
+              cases:
+                true: redirect
+                false: article
+          - id: url
+            type: strz
+            doc: string with the URL as refered in the URL pointer list
+          - id: title
+            type: strz
+            doc: string with an title as refered in the Title pointer list or empty; in case it is empty, the URL is used as title
+          - id: parameter
+            size: parameter_size
+            doc: (not used) extra parameters
+        instances:
+          mime:
+            value: _root.mime_list.items[mime_idx]
+            if: mime_idx != 0xffff
+
+        types:
+          article:
+            seq:
+              - id: cluster_index
+                -orig-id: cluster_number
+                type: u4
+                doc: cluster number in which the data of this directory entry is stored
+              - id: blob_index
+                -orig-id: blob_number
+                type: u4
+                doc: blob number inside the compressed cluster where the contents are stored
+            instances:
+              cluster:
+                value: _root.cluster_pointer_list[cluster_index].cluster
+              data:
+                value: cluster.blobs[blob_index]
+          redirect:
+            seq:
+              - id: redirect_index
+                -orig-id: redirect_index
+                type: u4
+                doc: pointer to the directory entry of the redirect target
+        enums:
+          namespace:
+            0x2d: #'-'
+              id: layout
+              doc: eg. the LayoutPage, CSS, favicon.png (48x48), JavaScript and images not related to the articles
+            0x41: #'A'
+              id: article
+              doc: articles - see Article Format
+            0x42: #'B'
+              id: article_meta_data
+              doc: article meta data - see Article Format
+            0x49: #'I'
+              id: image_file
+              doc: images, files - see Image Handling
+            0x4A: #'J'
+              id: image_text
+              doc: images, text - see Image Handling
+            0x4d: #'M'
+              id: meta_data
+              doc: ZIM metadata - see Metadata
+            0x55: #'U'
+              id: category_text
+              doc: categories, text - see Category Handling
+            0x56: #'V'
+              id: category_article_list
+              doc: categories, article list - see Category Handling
+            0x57: #'W'
+              id: category_list_per_article
+              doc: categories per article, category list - see Category Handling
+            0x58: #'X'
+              id: fulltext_index
+              doc: fulltext index - see ZIM Index Format
```

Comment:
Actually there is also 'Z', which is not documented, but is for 'Xapian indexes':

https://github.com/openzim/libzim/blob/master/src/search.cpp#L267

I have no idea what these do though ;-)

Reply:
Thanks for the info!

Reply:
Added.